config/v1/types_ingress.go

```diff
@@ -152,8 +153,53 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is present only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// You can assign Elastic IP addresses to the Network Load Balancer.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	// The EIPs in the cluster config are only used when the operator creates the default IngressController.
+	// It isn't used as a default value when eipAllocations is not specified on the IngressController.
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="EIP allocations cannot contain duplicates"
+	// +kubebuilder:validation:MaxItems=10
+	EIPAllocations []EIPAllocation `json:"eipAllocations"`
+}
+
+// EIPAllocation is a reference to available eip allocations in AWS environment.
+// The pattern for an EIP allocation ID is:
+// Starts with eipalloc-
+// Followed by an alphanumeric string of exactly 17 characters
+// Explanation of the regex `^eipalloc-[0-9a-fA-F]{17}$` for validating value of the EIPAllocation.
+// ^eipalloc- ensures the string starts with "eipalloc-".
+// [0-9a-fA-F]{17} matches exactly 17 hexadecimal characters (0-9, a-f, A-F).
+// $ ensures the string ends after the 17 hexadecimal characters.
+// Example of Valid and Invalid values:
+// eipalloc-1234567890abcdef1 is valid.
+// eipalloc-1234567890abcde is not valid (too short).
+// eipalloc-1234567890abcdefg is not valid (contains a non-hex character 'g').
+// Max length is calculated as follows:
+// eipalloc- = 9 chars and 17 hexadecimal chars after `-`
+// So, total is 17 + 9 = 26 chars required for value of an EIPAllocation.
+// +kubebuilder:validation:MinLength=26
+// +kubebuilder:validation:MaxLength=26
+// +kubebuilder:validation:XValidation:rule=`!self.contains(',')`,message="eipAllocations cannot contain a comma"
```

Comment:
@gcs278 we discussed to to split regex for having more specific messages. https://github.com/openshift/api/pull/1826#discussion_r1650785126

Reply:
Ack. I suppose that's fine, but it does seem a bit arbitrary to me.

All other special characters, e.g. `:;[]{}!@#$%^&*()`, are also invalid as a EIPAllocation ID, but we don't call out each one of those. Comma getting it's own special error message seems a bit incongruent. I suppose you are calling out comma specifically because it breaks the annotation delimitation, but I wonder if that really matters as it isn't valid as a EIPAllocation ID anyways.

It's a nit pick, not a big deal. I'm just trying to understand the value added for an error message for this specific character.

Reply:
But wont it help user identify the exact issue ?

Reply:
It would, but, by that argument you would also then want the same for any other special character. I don't think it's necessary personally and would drop it